File: sdks/dotnet/src/Entity.cs
namespace NovelCraft.Sdk;

internal class Entity : IEntity {
  public IOrientation Orientation { get; set; } = new Orientation(0, 0);
  public IPosition<decimal> Position { get; set; } = new Position<decimal>(0, 0, 0);
  public int TypeId { get; }
  public int UniqueId { get; }


  public Entity(int uniqueId, int typeId, IPosition<decimal> position, IOrientation orientation) {
    UniqueId = uniqueId;
    TypeId = typeId;
    Position = position;
    Orientation = orientation;
  }
}

File: sdks/c/src/entity.c
#include "internal/entity.h"

#include <stdlib.h>

#include "orientation.h"
#include "position.h"

ncsdk_Entity *ncsdk_Entity_New(ncsdk_Orientation orientation,
                               ncsdk_Position(float) position, int type_id,
                               int unique_id) {
  ncsdk_Entity *entity = malloc(sizeof(ncsdk_Entity));
  entity->orientation = orientation;
  entity->position = position;
  entity->type_id = type_id;
  entity->unique_id = unique_id;
  return entity;
}

void ncsdk_Entity_Delete(ncsdk_Entity *entity) { free(entity); }

ncsdk_Orientation ncsdk_Entity_GetOrientation(const ncsdk_Entity *entity) {
  return entity->orientation;
}

ncsdk_Position(float) ncsdk_Entity_GetPosition(const ncsdk_Entity *entity) {
  return entity->position;
}

int ncsdk_Entity_GetTypeId(const ncsdk_Entity *entity) {
  return entity->type_id;
}

int ncsdk_Entity_GetUniqueId(const ncsdk_Entity *entity) {
  return entity->unique_id;
}
